Applications/Utils/MeshEdit/CreateBoundaryConditionsAlongPolylines.cpp

```diff
@@ -0,0 +1,250 @@
+#include "GeoLib/PointWithID.h"
+
+// MeshLib
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+
+// MeshGeoToolsLib
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+
+void convertMeshNodesToGeometry(std::vector<MeshLib::Node*> const& nodes,
+	std::vector<std::size_t> const& node_ids,
+	std::string & geo_name,
+	GeoLib::GEOObjects & geometry_sets)
+{
+	// copy data
+	std::vector<GeoLib::Point*> * pnts(new std::vector<GeoLib::Point*>);
+	std::map<std::string, std::size_t>* pnt_names(
+		new std::map<std::string, std::size_t>);
+	std::size_t cnt(0);
+	for (std::size_t id: node_ids) {
+		pnts->push_back(new GeoLib::Point(nodes[id]->getCoords()));
+		pnt_names->insert(std::pair<std::string, std::size_t>(
+			geo_name+"-PNT-"+std::to_string(cnt), cnt));
+		cnt++;
+	}
+
+	// create data structures for geometry
+	geometry_sets.addPointVec(pnts, geo_name, pnt_names);
+}
+
+void writeBCsAndGML(GeoLib::GEOObjects & geometry_sets,
+	std::string & geo_name)
+{
+	INFO("write points to \"%s.gml\".", geo_name.c_str());
+	FileIO::BoostXmlGmlInterface xml_io(geometry_sets);
+	xml_io.setNameForExport(geo_name);
+	xml_io.writeToFile(geo_name+".gml");
+
+	GeoLib::PointVec const* pnt_vec_objs(geometry_sets.getPointVecObj(geo_name));
+	std::vector<GeoLib::Point*> const& pnts(*(pnt_vec_objs->getVector()));
+	std::string fname(geo_name+".gli");
+	std::ofstream out (fname.c_str());
+	out << "#POINTS\n";
+	out.precision(20);
+	std::ofstream bc_out (geo_name+".bc");
+	for (std::size_t k(0); k<pnts.size(); k++) {
+		out << k << " " << *(pnts[k]);
+		std::string pnt_name;
+		if (pnt_vec_objs->getNameOfElementByID(k, pnt_name)) {
+			out << "$NAME " << pnt_name;
+			bc_out << "#BOUNDARY_CONDITION\n";
+			bc_out << "  $PCS_TYPE\n";
+			bc_out << "    LIQUID_FLOW\n";
+			bc_out << "  $PRIMARY_VARIABLE\n";
+			bc_out << "    PRESSURE1\n";
+			bc_out << "  $GEO_TYPE\n";
+			bc_out << "    POINT " << pnt_name << "\n";
+			bc_out << "  $DIS_TYPE\n";
+			bc_out << "    CONSTANT 0.0\n";
+		}
+		out << "\n";
+	}
+	out << "#STOP\n";
+	bc_out << "#STOP\n";
+	bc_out.close();
+	out.close();
+}
+
+int main (int argc, char* argv[])
+{
+	LOGOG_INITIALIZE();
+	logog::Cout* logog_cout (new logog::Cout);
+	BaseLib::LogogSimpleFormatter *custom_format (new BaseLib::LogogSimpleFormatter);
+	logog_cout->SetFormatter(*custom_format);
+
+	TCLAP::CmdLine cmd(
+		"Creates boundary conditions for mesh nodes along polylines.",
+		' ',
+		"0.1");
+	TCLAP::ValueArg<std::string> mesh_arg("m", "mesh-file",
+		"the name of the file containing the mesh", true,
+		"", "file name");
+	cmd.add(mesh_arg);
+	TCLAP::ValueArg<std::string> geometry_fname("g", "geometry",
+		"the name of the file containing the input geometry", true,
+		"", "file name");
+	cmd.add(geometry_fname);
+	TCLAP::ValueArg<bool> vis_arg("v", "visualize",
+		"write gml file with found mesh nodes for visualization", false, 0, "bool");
+	cmd.add(vis_arg);
+	TCLAP::ValueArg<double> search_length_arg("s", "search-length",
```

Comment:
The default value for search length is `std::numeric_limits<double>::epsilon()`.